src/plugins/navigation_embeddable/public/components/dashboard_link/dashboard_link_component.tsx

```diff
@@ -6,55 +6,120 @@
          */
-        return await fetchDashboard(link.destination);
+        const dashboard = await fetchDashboard(link.destination)
+          .then((result) => {
+            setError(undefined);
+            return result;
+          })
+          .catch((e) => setError(e));
+        return dashboard;
       }
     }, [link, parentDashboardId]);
 
-  return (
-    <EuiButtonEmpty
-      isLoading={loadingDestinationDashboard}
-      iconType={NavigationLinkInfo[DASHBOARD_LINK_TYPE].icon}
-      {...(link.destination === parentDashboardId
-        ? {
-            color: 'text',
+  const [dashboardTitle, dashboardDescription] = useMemo(() => {
+    return link.destination === parentDashboardId
+      ? [parentDashboardTitle, parentDashboardDescription]
+      : [destinationDashboard?.attributes.title, destinationDashboard?.attributes.description];
+  }, [
+    link.destination,
+    parentDashboardId,
+    parentDashboardTitle,
+    destinationDashboard,
+    parentDashboardDescription,
+  ]);
+
+  const linkLabel = useMemo(() => {
+    return link.label || (dashboardTitle ?? DashboardLinkStrings.getDashboardErrorLabel());
+  }, [link, dashboardTitle]);
+
+  return loadingDestinationDashboard ? (
+    <li id={`dashboardLink--${link.id}--loading`}>
+      <EuiButtonEmpty size="s" isLoading={true}>
+        {DashboardLinkStrings.getLoadingDashboardLabel()}
+      </EuiButtonEmpty>
+    </li>
+  ) : (
+    <EuiListGroupItem
+      size="s"
+      color="text"
+      isDisabled={Boolean(error)}
+      id={`dashboardLink--${link.id}`}
+      iconType={error ? 'warning' : undefined}
+      iconProps={{ className: 'dashboardLinkIcon' }}
+      className={classNames('navigationLink', {
+        navigationLinkCurrent: link.destination === parentDashboardId,
+        dashboardLinkError: Boolean(error),
+        'dashboardLinkError--noLabel': !link.label,
+      })}
+      onClick={
+        link.destination === parentDashboardId
+          ? undefined
+          : () => {
+              // TODO: As part of https://github.com/elastic/kibana/issues/154381, connect to drilldown
+            }
+      }
+      label={
+        <EuiToolTip
+          delay="long"
+          display="block"
+          repositionOnScroll
+          position={layout === NAV_VERTICAL_LAYOUT ? 'right' : 'bottom'}
+          title={
+            error
+              ? DashboardLinkStrings.getDashboardErrorLabel()
+              : dashboardDescription
+              ? dashboardTitle
+              : undefined
           }
-        : {
-            color: 'primary',
-            onClick: () => {}, // TODO: As part of https://github.com/elastic/kibana/issues/154381, connect to drilldown
-          })}
-    >
-      {link.label ||
-        (link.destination === parentDashboardId
-          ? parentDashboardTitle
-          : destinationDashboard?.attributes.title)}
-    </EuiButtonEmpty>
+          content={error ? error.message : dashboardDescription || dashboardTitle}
```

Comment:
Similar to https://github.com/elastic/kibana/pull/162285#discussion_r1287525000 - because `dashboardDescription` can be the empty string, I think `||` works better here so you don't end up with `''` for the description 🤔 What do you think?

Reply:
Oh you're right, nevermind. Your solution is good.